fuel/converters/iris.py

```diff
@@ -0,0 +1,71 @@
+import os
+
+import h5py
+import numpy
+
+from fuel.converters.base import fill_hdf5_file
+
+
+def convert_iris(directory, output_directory, output_filename='iris.hdf5'):
+    """Converts the Iris dataset to HDF5.
+    Converts the Iris dataset to an HDF5 dataset compatible with
+    :class:`fuel.datasets.Iris`. The converted dataset is
+    saved as 'iris.hdf5'.
+    This method assumes the existence of the file `iris.data`.
+    It assumes the existence of the following file:
+    * `iris.data`
+    Parameters
+    ----------
+    directory : str
+        Directory in which input files reside.
+    output_directory : str
+        Directory in which to save the converted dataset.
+    output_filename : str, optional
+        Name of the saved dataset. Defaults to `None`, in which case a name
+        based on `dtype` will be used.
+    dtype : str, optional
+        Either 'float32', 'float64', or 'bool'. Defaults to `None`,
+        in which case images will be returned in their original
+        unsigned byte format.
+    Returns
+    -------
+    output_paths : tuple of str
+        Single-element tuple containing the path to the converted dataset.
+    """
+
+    output_path = os.path.join(output_directory, output_filename)
+    h5file = h5py.File(output_path, mode='w')
+    classes = {'Iris-setosa': 0, 'Iris-versicolor': 1, 'Iris-virginica': 2}
+    data = numpy.loadtxt(
+        os.path.join(directory, 'iris.data'),
+        converters={4: lambda x: classes[x]},
+        delimiter=',')
+    numpy.random.shuffle(data)
+    features = data[:, :-1].astype('float32')
+    targets = data[:, -1].astype('uint8').reshape((-1, 1))
+    train_features = features[:100]
+    train_targets = targets[:100]
+    valid_features = features[100:120]
+    valid_targets = targets[100:120]
+    test_features = features[120:]
+    test_targets = targets[120:]
```

Comment:
I don't think there's a canonical split for Iris, right? In that case, we should just make one `'all'` split and let users manually select their desired split with `subset` or through an iteration scheme.